src/Ads/AssetSuggestionsService.php

```diff
@@ -111,10 +122,70 @@ public function __construct( WP $wp, WC $wc, ImageUtility $image_utility, wpdb $
 	 * @param string $type Only possible values are post or term.
 	 */
 	public function get_assets_suggestions( int $id, string $type ): array {
-		// TODO: Fetch assets from others campaigns and merge the result with the WP Assets.
+		$asset_group_assets = $this->get_asset_group_asset_suggestions( $id, $type );
+
+		if ( ! empty( $asset_group_assets ) ) {
+			return $asset_group_assets;
+		}
+
 		return $this->get_wp_assets( $id, $type );
 	}
 
+	/**
+	 * Get URL for a specific post or term.
+	 *
+	 * @param int    $id Post or Term ID.
+	 * @param string $type Only possible values are post or term.
+	 *
+	 * @return string The URL.
+	 * @throws Exception If the ID is invalid.
+	 */
+	protected function get_url( int $id, string $type ): string {
+		if ( $type === 'post' ) {
+			$url = get_permalink( $id );
+		} else {
+			$url = get_term_link( $id );
+		}
+
+		if ( $url === false || is_wp_error( $url ) ) {
+			throw new Exception(
+				/* translators: 1: is an integer representing an unknown Term ID */
+				sprintf( __( 'Invalid Term ID or Post ID %1$d', 'google-listings-and-ads' ), $id )
+			);
+		}
+
+		return $url;
+
+	}
+
+
+
+	/**
+	 * Get other campaigns' assets from the specific url.
+	 *
+	 * @param int    $id Post or Term ID.
+	 * @param string $type Only possible values are post or term.
+	 */
+	protected function get_asset_group_asset_suggestions( int $id, string $type ): array {
+		$final_url = $this->get_url( $id, $type );
+
+		$assets = array_values( $this->asset_group_asset->get_assets_by_final_url( $final_url ) );
+
+		// Suggest the assets from the first asset group if exists.
+		$asset_group_assets = array_shift( $assets );
```

Comment:
> We don't really need to remove the indexes with array_values if all we are going to do is shift the first value off the array. My personal preference is also to use array_reset since it won't change the original array or trigger a reindex. Probably negligible in this case though.

Thanks, yes array `reset` is a better option here.

> Looking at this though if we are only interested in the first set of assets then couldn't we make that the responsibility of AdsAssetGroupAsset with a function like get_first_asset_group_assets_by_final_url? Or even just leaving it as get_assets_by_final_url is fine but then clarify in the docblock that it's getting the assets from the first asset group it encounters.

> If we feel like we are going to reuse this more often we could create a specific helper (like [get_row](https://github.com/woocommerce/google-listings-and-ads/blob/2.3.5/src/DB/QueryInterface.php#L77-L82)) function in Query to only fetch the first result. This can be done by setting the [pageSize to 1](https://github.com/woocommerce/google-listings-and-ads/blob/2.3.5/src/API/Google/Query/ReportQueryTrait.php#L50). Or alternatively we can just process the first $row and return early, so we don't have to go through the process of converting all the assets.

I could move the responsibility however `get_assets_by_final_url ` will still need to process all assets because the query returns an asset for each `row` (each `row` contains and `asset` and `asset_group`) , so in this case using `pageSize = 1` or processing the first `row` would not work. That was the reason why I leave the responsibility to get the first asset group to the function `get_asset_group_asset_suggestions`. 

Another alternative, if we would like to move the responsibility, could be to add a param to the function `get_assets_by_final_url ` something like `get_assets_by_final_url( $url, $only_first_asset_group)`{ , so it is clear that we are specifically getting the assets from the first asset group.  WDYT?

Reply:
Thanks for the clarification I see what you mean now, they are all being grouped by the asset group ID's and could be in a random order. I think the `$only_first_asset_group` option makes sense in this case.
That can either be implemented by then returning only the first array result or converting only the assets that match the first group we encounter (second option would be less processing, but probably negligible since we already fetched all assets anyway).